### PR TITLE
Natasha/listening

### PR DESCRIPTION
**Pull request by natashaarmbrust**
- add "Jump Back In" feature in `BookmarkViewController`
- add dismissing Facebook friends 

Right now I'm waiting on backend for the full jump back in feature integration. The UI works now it just doesn't show actually dismiss when you refresh the view.

### CODE REVIEW

File: Podcast/BookmarkViewController.swift [hunk 1]
```diff
@@ -158,3 +212,63 @@ class BookmarkViewController: ViewController, EmptyStateTableViewDelegate, UITab
...
+            self.continueListeningCollectionView.reloadData()
+            self.bookmarkTableView.reloadData()
+            self.layoutHeaderView()
+        }
+        endpointRequest.failure = { _ in
+        }
+        System.endpointRequestQueue.addOperation(endpointRequest)
+    }
+
+    func continueListeningCollectionViewCellDismissButtonPress(cell: ContinueListeningCollectionViewCell) {
```
Comment (doctor3w) [inline Podcast/BookmarkViewController.swift]: dismissButtonPressed(on continueListeningCell: ..)

File: Podcast/ContinueListeningCollectionViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+//  Podcast
+//
+//  Created by Natasha Armbrust on 3/29/18.
+//  Copyright © 2018 Cornell App Development. All rights reserved.
+//
+
+import UIKit
+
+protocol ContinueListeningCollectionViewCellDelegate: class {
+    func continueListeningCollectionViewCellDismissButtonPress(cell: ContinueListeningCollectionViewCell)
```
Comment (doctor3w) [inline Podcast/ContinueListeningCollectionViewCell.swift]: like I said above.

File: Podcast/FacebookFriendsTableViewCell.swift [hunk 3]
```diff
@@ -12,6 +12,7 @@ protocol FacebookFriendsTableViewCellDelegate: class {
     func facebookFriendsTableViewCellDidPressFollowButton(tableViewCell: FacebookFriendsTableViewCell, collectionViewCell: FacebookFriendsCollectionViewCell, indexPath: IndexPath)
     func facebookFriendsTableViewCellDidSelectRowAt(tableViewCell: FacebookFriendsTableViewCell, collectionViewCell: FacebookFriendsCollectionViewCell, indexPath: IndexPath)
     func facebookFriendsTableViewCellDidPressSeeAllButton(tableViewCell: FacebookFriendsTableViewCell)
+    func facebookFriendsTableViewCellDidPressDismissButton(tableViewCell: FacebookFriendsTableViewCell, collectionViewCell: FacebookFriendsCollectionViewCell, indexPath: IndexPath)
```
Comment (doctor3w) [inline Podcast/FacebookFriendsTableViewCell.swift]: haha maybe change all these too if you get the chance. 
facebookFriendsTableView(didPress action: [NEW_ACTION_ENUM], on cell: CellType)
something like that. Can make the actions into an enum, makes it easy to add actions. 
Or at least rename and keep separate.

**Review by doctor3w — CHANGES_REQUESTED** [review]
Just a few naming things!

File: Podcast/ContinueListeningCollectionViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func configure(for episode: Episode) {
+        artworkImageView.setImageAsynchronouslyWithDefaultImage(url: episode.largeArtworkImageURL)
+        titleLabel.text = episode.title
+        descriptionLabel.text = episode.getDateTimeLabelString(includeSeriesTitle: true)
+        if let duration = Double(episode.duration), episode.isDurationWritten {
+            let minsLeft = Int((duration - episode.currentProgress) / 60)
+            timeLeftLabel.text = "\(minsLeft) min left"
```
Comment (mindylou) [inline Podcast/ContinueListeningCollectionViewCell.swift]: what if there are hours left?

File: Podcast/DismissFacebookFriendEndpointRequest.swift [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+        super.init()
+
+        path = "/users/facebook/friends/ignore/\(facebookId)/"
+        httpMethod = .post
+        headers = ["AccessToken": facebookAccessToken]
+    }
+
+    override func processResponseJSON(_ json: JSON) {
+        print(json)
```
Comment (mindylou) [inline Podcast/DismissFacebookFriendEndpointRequest.swift]: remove print

File: Podcast/FacebookFriendsCollectionViewCell.swift [hunk 6]
```diff
@@ -68,6 +70,13 @@ class FacebookFriendsCollectionViewCell: UICollectionViewCell {
         button.addTarget(self, action: #selector(didPressFollowButton), for: .touchUpInside)
         addSubview(button)
 
+        dismissButton = Button()
+        dismissButton.setImage(#imageLiteral(resourceName: "dismiss_icon_light"), for: .normal)
+        dismissButton.imageEdgeInsets = UIEdgeInsets.zero
```
Comment (mindylou) [inline Podcast/FacebookFriendsCollectionViewCell.swift]: you can do `.zero` here

File: Podcast/ContinueListeningCollectionViewCell.swift (hunk 4 above)
Comment (mindylou) [reply Podcast/ContinueListeningCollectionViewCell.swift]: image
image
I don't think these are matching up based on the episode current progress

**Pull request merged**